### PR TITLE
Device Drag Pull request

### PR DESCRIPTION
**Pull request by sigh-gone**
This pull request is for allowing users to drag the devices to different spots.

### CODE REVIEW

File: src/schematic.rs [hunk 1]
```diff
@@ -29,9 +29,10 @@ use iced::{
 };
 use nets::{NetEdge, NetVertex, Nets};
 use std::sync::Arc;
-use std::{collections::HashSet, fs};
+use std::{collections::HashSet, fs, process};
 
 use colored::Colorize;
+use iced_aw::Icon::Command;
```
Comment (ua-kxie) [inline src/schematic.rs:35]: ?

File: src/schematic.rs [hunk 2]
```diff
@@ -186,11 +187,11 @@ impl Default for Schematic {
         #[cfg(target_os = "macos")]
         {
             // retrieve libngspice.dylib from the following possible directories
-            let ret = Cmd::new("find")
+            let ret = process::Command::new("find")
```
Comment (ua-kxie) [inline src/schematic.rs:190]: ?

File: src/schematic.rs [hunk 3]
```diff
@@ -186,11 +187,11 @@ impl Default for Schematic {
...
                 .args(&["/usr/lib", "/usr/local/lib"])
                 .arg("-name")
                 .arg("*libngspice.dylib")
-                .stdout(Stdio::piped())
+                .stdout(process::Stdio::piped())
```
Comment (ua-kxie) [inline src/schematic.rs:193]: this is my bad - I didn't realize process::* it was being used in here - feel free to import it however you like

File: src/schematic.rs [hunk 4]
```diff
@@ -762,7 +764,28 @@ impl Schematic {
                 SchematicState::Idle,
                 Event::Mouse(iced::mouse::Event::ButtonPressed(iced::mouse::Button::Left)),
             ) => {
-                state = SchematicState::Selecting(SSBox::new(curpos_ssp, curpos_ssp));
+                let mut click_selected = false;
+
+                if !&self.selected.is_empty() {
```
[the remark this answers is not in the document]
Comment (sigh-gone) [reply src/schematic.rs:768]: okay, I'll removed, it wasn't sure

**Pull request merged**